core/src/main/java/org/commonjava/indy/core/content/DefaultDownloadManager.java

```diff
@@ -537,8 +547,9 @@ public Transfer store( final ArtifactStore store, final String path, final Input
         {
             final HostedRepository deploy = (HostedRepository) store;
 
-            final ArtifactPathInfo pathInfo = ArtifactPathInfo.parse( path );
-            if ( pathInfo != null && pathInfo.isSnapshot() )
+//            final ArtifactPathInfo pathInfo = ArtifactPathInfo.parse( path );
+            final ContentQuality quality = contentAdvisor.getContentQuality( path );
```

Comment:
Once we have support for multiple ContentAdvisors, we will need to iterate through them and accept the first non-null result for quality. It's conceivable that might lead to advisor ordering problems, but that's something we could manage using a mechanism for sorting them during init (a problem to be solved later).